src/test/java/uk/gov/justice/log/integration/SearchServiceIT.java

```diff
@@ -0,0 +1,681 @@
+        File dataDir = new File(datadir);
+        if (dataDir.exists()) {
+            FileSystemUtils.deleteSubDirectories(Paths.get(datadir));
+        }
+    }
+
+    public static RestClient restClient() {
+        return RestClient.builder(new HttpHost(restConfig.getHostName(),
+                restConfig.getHostPort(), restConfig.getHostScheme()))
+                .setMaxRetryTimeoutMillis(MAX_RETRY_TIMEOUT_MILLIS)
+                .setRequestConfigCallback(new RequestConfigBuilderWrapper(restConfig))
+                .setHttpClientConfigCallback(new HttpAsyncClientBuilderWrapper(new ConnectionManager()))
+                .build();
+    }
+
+    private static void uploadSampleData(String path) throws IOException {
+        byte[] encoded = Files.readAllBytes(Paths.get(SAMPLE_LOGS_PATH));
+        final String logs = new String(encoded, "UTF-8");
+
+        final String endpoint = ELASTIC_SEARCH_CLUSTER_URL + "_bulk?pretty";
+        final HttpEntity data = new NStringEntity(logs, APPLICATION_JSON);
+        restClient().performRequest(POST, endpoint, Collections.emptyMap(), data);
+    }
+
+    private static void mockSetupForConfig(final String hostName, final String hostScheme, final Integer hostPort,
+                                           final int proxyPort, final String proxyHost) throws IOException {
+
+        final Map<String, Object> data = new HashMap<>();
+        data.put("hostName", hostName);
+        data.put("hostScheme", hostScheme);
+        data.put("hostPort", hostPort);
+        data.put("proxyPort", proxyPort);
+        data.put("proxyHost", proxyHost);
+
+        new Yaml().dump(data, new FileWriter(CONFIG_PATH));
+    }
+
+    private static void mockSetupForSearchCriteria(final List<String> keywords, final List<String> regexes,
+                                                   final int durationMinutes, final String fromTime, final String toTime) throws IOException {
+        final Map<String, Object> data = new HashMap<>();
+        data.put("keywords", keywords);
+        data.put("regexes", regexes);
+        data.put("durationMinutes", durationMinutes);
+        if (!StringUtils.isEmpty(fromTime)) {
+            data.put("fromTime", fromTime);
+        }
+        if (!StringUtils.isEmpty(toTime)) {
+            data.put("toTime", toTime);
+        }
+        new Yaml().dump(data, new FileWriter(SEARCH_PATH));
+    }
+
+    private static JsonObject mappings() {
+
+        final JsonObject type = Json.createObjectBuilder().add("type", "keyword").build();
+        final JsonObject productId = Json.createObjectBuilder().add("message", type).build();
+        final JsonObject properties = Json.createObjectBuilder().add("properties", productId).build();
+        final JsonObject products = Json.createObjectBuilder().add("elasticsearch", properties).build();
+        final JsonObject mappings = Json.createObjectBuilder().add("mappings", products).build();
```

Comment:
This can be done with nested builders without calling build() on each object. It would make it easier to see the structure being created.